### PR TITLE
chore: Adjust upgrade guide

### PR DESCRIPTION
**Pull request by tomfrenken**
This PR adjusts the upgrade to give clearer instructions and improves some grammar.

Closes SAP/cloud-sdk-backlog#1065

- [x] I know which base branch I chose for this PR, as the default branch is `v2-main` now, which is not for v3 development.
- [x] If my change will be merged into the `main` branch (for v3), I've updated (V3-Upgrade-Guide.md)[./V3-Upgrade-Guide.md] in case my change has any implications for users updating to SDK v3

### CODE REVIEW

File: V3-Upgrade-Guide.md [hunk 1]
```diff
@@ -81,23 +81,20 @@ The following sub-sections describe affected modules, functions and interfaces w
...
   .execute({
-    destinationName:
-    'DESTINATION',
+    destinationName: 'DESTINATION',
     useCache: true,
     isolationStrategy: 'tenant'
   })
   ```
 
-- The generic type parameter `JwtKeysT` in `JwtKeyMapping` is now narrowed to extend `string`.
```
Comment (fwilhe) [inline V3-Upgrade-Guide.md:91]: why delete this line?

Comment (tomfrenken) [reply V3-Upgrade-Guide.md:91]: Because it is already there a few lines above

**Comment by fwilhe** [comment]
My original motivation which moved into this ticket was that I was missing guidance on how to fix issues with renamed identifiers such as `Destination` to `HttpDestination`. I still don't see any mention of this in the document. We have a line saying that the `Destination` type as changed, but imo it does not really properly tell me what I need to do.

We don't need too much handholding, but something like "Renaming instances of Destination to HttpDestination` is something we should have in the guide, imo.

Not sure if we have more cases where this would apply. My hope would be that with the guide a developer who is unaware of sdk internals has enough information to fix issues caused by the upgrade without needing to google or guess.

**Comment by FrankEssenberger** [comment]
> My original motivation which moved into this ticket was that I was missing guidance on how to fix issues with renamed identifiers such as `Destination` to `HttpDestination`. I still don't see any mention of this in the document. We have a line saying that the `Destination` type as changed, but imo it does not really properly tell me what I need to do.
> 
> We don't need too much handholding, but something like "Renaming instances of Destination to HttpDestination` is something we should have in the guide, imo.
> 
> Not sure if we have more cases where this would apply. My hope would be that with the guide a developer who is unaware of sdk internals has enough information to fix issues caused by the upgrade without needing to google or guess.

I think it would be best to add things you find missing also to this PR.

**Review by FrankEssenberger — APPROVED** [review]
The changes are fine for me - I agree with Florian that in many places the guide could be a bit more elaborate what to do.

**Comment by tomfrenken** [comment]
Yeah, I wasn't sure at what point I explain too much but I guess I can add more instructions

**Pull request merged**